card/db/migrate/20150220141031_create_deck_routes_file.rb

```diff
@@ -0,0 +1,14 @@
+# -*- encoding : utf-8 -*-
+
+class CreateDeckRoutesFile < ActiveRecord::Migration
+  def up
+    routes_path = File.join Wagn.root, 'config', 'routes.rb'
+    if !File.exist? routes_path
+      template_path = File.join Wagn.gem_root, 'lib', 'wagn', 'generators', 'wagn', 'templates', 'config', 'routes.erb'
+      @include_jasmine_engine = false
+      decko_path = ''
+      routes = ERB.new(File.read template_path).result binding
+      File.open(routes_path, "w") { |f|  f.write routes }
+    end
+  end
+end
```

Comment:
I think you meant that reverse, but yes.  With nothing in config/application.rb, wagn/application.rb tells it to find it in the gem, but you can override to get it from the deck instead of the gem.  The engine deck case will not use wagn/application.rb and get the standard config/routes.rb where the developer has added the Decko::Engine mount.  If this gets any significant use, then a shortcut might be good.

And, in the engine deck case, there still is a config/application.rb, but we don't generate it, and it is for the hosting application.  Now, most of application.rb is comments about config settings that you could change, and those might be useful for engine developers too.  I'm not worried about that now, although it might be nice to have all those comments somewhere so they could add them manually if desired.